ozaria/site/components/interactive/PageInteractive/insertCode/index.vue

```diff
@@ -142,13 +165,55 @@
 
         const lineToReplace = this.localizedInteractiveConfig.lineToReplace - 1
 
-        if (this.selectedAnswer.id !== -1) {
+        if (this.questionAnswered) {
           this.codemirror.addLineClass(lineToReplace, 'background', 'highlight-line-answered')
           this.codemirror.removeLineClass(lineToReplace, 'background', 'highlight-line-prompt')
         } else {
           this.codemirror.addLineClass(lineToReplace, 'background', 'highlight-line-prompt')
           this.codemirror.removeLineClass(lineToReplace, 'background', 'highlight-line-answered')
         }
+      },
+
+      async submitSolution () {
+        if (!this.questionAnswered) {
+          return
+        }
+
+        this.showModal = true
+
+        // TODO save through vuex and block progress until save is successful
+        await putSession(this.interactive._id, {
+          json: {
+            codeLanguage: this.codeLanguage,
+            submission: this.solution
+          }
+        })
+      },
+
+      closeModal () {
+        if (this.solution.correct) {
+          this.$emit('completed')
+        } else {
+          this.resetAnswer()
+          this.updateHighlightedLine()
+        }
+
+        this.showModal = false
+      },
+
+      stateFromCompleteSession () {
+      //   TODO should answer be recovered or should use play as if they have never played?
```

Comment:
My understanding was that revisiting an interactive that was successfully completed should be treated as "unplayed."  Is this correct?

Reply:
I was under the assumption that if it is completed it should pre-fill with the correct answer. In all other cases treat as if it's the first time.
But @shubhi1092 will know precise answer.

Reply:
Yeah if it is completed successfully, it should pre-fill with the correct answer. Thats true for all types of interactives

Reply:
Ok going to follow up with some clarifying questions and handle this in a follow up PR.  I've added to UX task to track.

Reply:
Came up with some additional questions upon implementing insert code so this PR will contain insert code implementation of state recovery.

Reply:
What is status of commented code below?

Reply:
Looks like this has been implemented in `selectedAnswer`, does this commented code needs to be removed?

Reply:
Yeah, removed